tools/tutorial_videos/tutorial_videos/session.py

```
@@ -0,0 +1,155 @@
+
+
+class XvfbDisplay:
+    """``Xvfb :N`` sized to the app window so the capture is exactly the app."""
+
+    def __init__(self, display: str = ":99", size: str = "1920x1080") -> None:
+        self.display = display
+        self.size = size
+        self._process: subprocess.Popen[bytes] | None = None
+
+    def __enter__(self) -> "XvfbDisplay":
+        self._process = subprocess.Popen(
```

Ruff (0.15.21)

[error] `subprocess` call: check for execution of untrusted input

(S603)